src/autoqasm/types/types.py

```diff
@@ -114,7 +115,7 @@ def __init__(
         if not isinstance(init_expression, Iterable):
             raise errors.InvalidArrayDeclaration("init_expression must be an iterable type.")
 
-        dimensions = [len(init_expression)]
+        dimensions = np.shape(init_expression)
```

Comment:
Oops! Indeed, for multi-dimensional arrays, we want to know the no. of entries in each dimension. As I understand, earlier, it worked because the test I added had a multi-dimensional array with `(2, 2)` dimensions, in which case `len(init_expression)` just worked fine, but not in general.

Reply:
With the previous change I saw the output actually as `array[int[32], 2] a = {{1, 2}, {3, 4}};`. Because `len(init_expression)` returned `2`, it was declaring an array with dimensions of `[2]` but using the 2-dimensional `init_expression`. This would have failed when trying to run.

But yes, as @jcjaskula-aws says, the test was missing an `assert` so the case was not actually being tested at all.